src/perlin.js

```diff
@@ -381,5 +381,48 @@
     var nz = module.fbm3(x + 7.1, y + 4.9, z + 1.3, o);
     return module.fbm3(x + wa * nx, y + wa * ny, z + wa * nz, o, f, l);
   };
+
+  function hash(x, y) {
+    x = x & 255;
+    y = y & 255;
+    //return new Grad(Math.abs(Math.sin(x*123.657+y*46.87)*423548.15763) % 1, Math.abs(Math.sin(x*87.391+y*157.597)*587654.26461)%1, 0);
+    return new Grad(perm[x + perm[y]] / 255.0, perm[y + perm[x]] / 255.0, 0);
+  }
```

Comment:
We should use an actual hash function instead of one we made up. Any particular reason for changing it from what we had previously though?